src/reducer.js

```diff
@@ -0,0 +1,53 @@
+const initialRestaurant = {
+  newId: 100,
+  name: '',
+  category: '',
+  address: '',
+  restaurants: [],
+};
+
+export default function reducer(restaurant = initialRestaurant, action) {
+  if (action === undefined) {
+    return restaurant;
+  }
+
+  if (action.type === 'updateName') {
+    return {
+      ...restaurant,
+      name: action.payload.name,
+    };
+  }
+
+  if (action.type === 'updateCategory') {
+    return {
+      ...restaurant,
+      category: action.payload.category,
+    };
+  }
+
+  if (action.type === 'updateAddress') {
+    return {
+      ...restaurant,
+      address: action.payload.address,
+    };
+  }
+
+  if (action.type === 'addRestaurant') {
+    const {
+      newId, name, category, address, restaurants,
+    } = restaurant;
+
+    return {
+      ...restaurant,
+      newId: newId + 1,
+      restaurants: [...restaurants, {
+        id: newId, name, category, address,
+      }],
```

Comment:
```suggestion
      restaurants: [
        ...restaurants, 
        {
          id: newId, 
          name, 
          category, 
          address,
        }
      ],
```
인자가 많은 경우에는 줄바꿈과 간격을 줌으로써 더 읽게 쉽게 해주는게 좋을 것 같아요~